shotover/src/transforms/kafka/sink_cluster/mod.rs

```diff
@@ -484,6 +572,28 @@ impl KafkaSinkCluster {
         Ok(results)
     }
 
+    async fn route_to_first_node(
```

Comment:
Routing to the first node in the list could cause issues in the case where:
* send handshake message 1
* shared_nodes is updated and a new node is first in the list.
* send handshake message 2

In this case the second handshake message goes to a different node.
To avoid this issue we need to a pick a node, store its node id and then continue sending messages to the node with that id.

I'm happy for an issue to be raised and this be addressed in a follow up PR.